spec/ecs_logging/logger_spec.rb

```diff
@@ -100,5 +100,46 @@ module EcsLogging
         )
       end
     end
+
+    context "when running alongside the Elastic APM agent" do
+      before do
+        module ::ElasticAPM
+          def self.running?
+            true
+          end
+
+          def self.current_transaction
+            Struct.new(:id, :trace_id).new('abc', 'xyz')
+          end
+
+          def self.current_span
+            Struct.new(:id).new('def')
+          end
+        end
+      end
+
+      after do
+        Object.send(:remove_const, :ElasticAPM)
+        raise if defined?(ElasticAPM)
+      end
```

Comment:
You could also just use `allow` to have `ElasticAPM.running?` return true. And doubles for `current_transaction` and `current_span`. That way, you aren't making real changes on the `ElasticAPM` module that you have to clean up in an `after` block after the spec runs, as they are `rspec` features.
for example
`allow(ElasticAPM).to receive(:running?).and_return(true)`
`allow(ElasticAPM).to receive(:current_transaction).and_return(double(id: 'abc'))`
etc

Reply:
The module isn't defined in this library unless we depend on the agent and I don't think we need to. We could and perhaps gain a level og sort of API safety, but I think this is fine?

Reply:
That's a good point-- I didn't realize we didn't depend on the agent in this library. This approach makes sense then 👍🏼